### PR TITLE
cleanup: refactor cleanup to let olm delete resources

### PR DESCRIPTION
**Pull request by varshaprasad96**
**Description of the change:**
Refactor cleanup to delete crds and csv instead of individually
deleting the resources referenced in install plan. OLM applies
an ownerref on every namespaced resource to the CSV, and an owner
label on cluster scoped resources. OLM and kube garbage collector
will delete the resources referenced in csv.

**Motivation for the change:**
Rely on OLM to delete operator resources. 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: internal/olm/operator/uninstall.go [hunk 1]
```diff
@@ -234,3 +232,40 @@ func (u *Uninstall) getInstallPlanResources(ctx context.Context, installPlanKey
...
+	installedCSV := &v1alpha1.ClusterServiceVersion{}
+	if err := u.config.Client.Get(ctx, key, installedCSV); err != nil {
+		return nil, err
+	}
+
+	installedCSV.SetGroupVersionKind(schema.GroupVersionKind{
+		Kind:    csvKind,
+		Version: installedCSV.GroupVersionKind().Version,
+		Group:   installedCSV.GroupVersionKind().Group,
+	})
```
Comment (estroz) [inline internal/olm/operator/uninstall.go:251]:
```suggestion
	installedCSV.SetGroupVersionKind(v1alpha1.SchemeGroupVersion.WithKind(csvKind))
```

File: internal/olm/operator/uninstall.go [hunk 2]
```diff
@@ -234,3 +232,40 @@ func (u *Uninstall) getInstallPlanResources(ctx context.Context, installPlanKey
...
+	for _, resource := range csv.Status.RequirementStatus {
+		if resource.Kind == crdKind {
+			obj := &unstructured.Unstructured{}
+			obj.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   resource.Group,
+				Version: resource.Version,
+				Kind:    resource.Kind,
+			})
+			obj.SetName(resource.Name)
+			obj.SetNamespace(csv.GetNamespace())
```
Comment (estroz) [inline internal/olm/operator/uninstall.go:265]: CRDs don't need a namespace
```suggestion
```

File: internal/olm/operator/uninstall.go [hunk 3]
```diff
@@ -234,3 +232,40 @@ func (u *Uninstall) getInstallPlanResources(ctx context.Context, installPlanKey
 	}
 	return crds, csvs, others, nil
 }
+
+func (u *Uninstall) getInstalledCSV(ctx context.Context, subscription *v1alpha1.Subscription) (*v1alpha1.ClusterServiceVersion, error) {
```
Comment (estroz) [inline internal/olm/operator/uninstall.go:236]: Comments

File: internal/olm/operator/uninstall.go [hunk 4]
```diff
@@ -234,3 +232,40 @@ func (u *Uninstall) getInstallPlanResources(ctx context.Context, installPlanKey
...
+
+	installedCSV.SetGroupVersionKind(schema.GroupVersionKind{
+		Kind:    csvKind,
+		Version: installedCSV.GroupVersionKind().Version,
+		Group:   installedCSV.GroupVersionKind().Group,
+	})
+	return installedCSV, nil
+}
+
+func getCRDs(csv *v1alpha1.ClusterServiceVersion) (crds []controllerutil.Object) {
```
Comment (estroz) [inline internal/olm/operator/uninstall.go:255]: Comments

**Review by joelanford — APPROVED** [review]
/lgtm

**Pull request merged**